p2p/transport/webrtc/udpmux/packetqueue.go

```diff
@@ -0,0 +1,80 @@
+package udpmux
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	pool "github.com/libp2p/go-buffer-pool"
+)
+
+type packet struct {
+	buf []byte
+}
+
+var errTooManyPackets = errors.New("too many packets in queue; dropping")
+
+const maxPacketsInQueue = 128
+
+type packetQueue struct {
+	packetsMux sync.Mutex
+	packetsCh  chan struct{}
```

Comment:
That would mean keeping a channel of length 128 in memory thought. That's roughly 1 kB of memory for every WebRTC connection. Now you might argue that the WebRTC transport is so terribly inefficient (with the code that we currently have) that this doesn't matter, and I'd have to concede that point to you. But I had pretty much exactly the same construction in quic-go (a channel of length 128 in every connection to buffer incoming packets), and it did show up in profiles.

Reply:
> But I had pretty much exactly the same construction in quic-go (a channel of length 128 in every connection to buffer incoming packets), and it did show up in profiles.

This code likely behaves worst than a channel of length 128. I swapped the "benchmark" with a channel to prove this point: https://gist.github.com/MarcoPolo/f8b8cbd8819961ebbde04bf06a712783.

It's a weak proof, because it's not a very good benchmark. But it does serve to show how unperformant the current implementation is.

1 KB is a small price to pay for the simplicity and performance this would give us.

Reply:
That's convincing. At 1000 WebRTC connections, that's only 1 MB anyway.

To make the current solution perform well, we'd have to use a ring buffer to avoid allocs, and then the space saving would be more dubious (if you ramp up the buffer size once, it would stay there).

I'll change it to a channel, and we can still optimize this at some point in the future, if this ever becomes a problem.

Reply:
Done. The packet queue is gone! Lots of code deleted, I like it!